frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx

```diff
@@ -27,6 +49,54 @@ import {
 } from '../types';
 import { requireOperatorGroup } from './operator-group';
 import { PackageManifestList } from './package-manifest';
+import { deleteCatalogSourceModal } from './modals/delete-catalog-source-modal';
+import { disableDefaultSourceModal } from './modals/disable-default-source-modal';
+import { OperatorHubKind } from './operator-hub';
+
+const catalogSourceModelReference = referenceForModel(CatalogSourceModel);
+const DEFAULT_SOURCE_NAMESPACE = 'openshift-marketplace';
+
+const deleteModal = (kind: K8sKind, catalogSource: CatalogSourceKind): KebabOption => ({
+  ...Kebab.factory.Delete(kind, catalogSource),
+  callback: () => deleteCatalogSourceModal({ kind, resource: catalogSource }),
+});
+
+const disableSourceModal = (
+  kind: K8sKind,
+  operatorHub: OperatorHubKind,
+  sourceName: string,
+): KebabOption => ({
+  label: 'Disable',
+  callback: () => disableDefaultSourceModal({ kind, operatorHub, sourceName }),
+  accessReview: asAccessReview(kind, operatorHub, 'patch'),
+});
+
+const enableSource = (
+  kind: K8sKind,
+  operatorHub: OperatorHubKind,
+  sourceName: string,
+): KebabOption => ({
+  label: 'Enable',
+  callback: () => {
+    const currentSources = _.get(operatorHub, 'spec.sources', []);
+    const patch = [
+      {
+        op: _.isEmpty(currentSources) ? 'add' : 'replace',
```

Comment:
`add` will do a replace if the key already exists: https://tools.ietf.org/html/rfc6902#section-4.1

>  If the target location specifies an object member that does exist, that member's value is replaced.

```suggestion
        op: 'add',
```